### PR TITLE
Fix TLS certificate download

### PR DESCRIPTION
**Pull request by swaterkamp**
**Checklist**:

- [] Tests
- [X] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: gsa/src/web/pages/tlscertificates/component.js [hunk 1]
```diff
@@ -142,6 +144,8 @@ const mapStateToProps = (rootState, {id}) => {
   const detailsExportFileName = userDefaultsSelector.getValueByName(
     'detailsexportfilename',
   );
+  // console.log('Selector', tlsCertificateSelector, id);
```
Comment (bjoernricks) [inline gsa/src/web/pages/tlscertificates/component.js]:
```suggestion
```

File: gsa/src/web/pages/tlscertificates/component.js [hunk 2]
```diff
@@ -142,6 +144,8 @@ const mapStateToProps = (rootState, {id}) => {
...
+  // console.log('state2props', tlsCertificateSelector.getEntity(id));
```
Comment (bjoernricks) [inline gsa/src/web/pages/tlscertificates/component.js]:
```suggestion
```

File: gsa/src/web/pages/tlscertificates/component.js [hunk 3]
```diff
@@ -57,34 +57,36 @@ class TlsCertificateComponent extends React.Component {
...
-      resourceName: name,
-      resourceType: entityType,
-      username,
-    });
-
-    onDownloaded({
-      filename,
-      mimetype: 'application/x-x509-ca-cert',
-      data: create_pem_certificate(certificate),
+    this.props.loadEntity(cert.id).then(response => {
```
Comment (bjoernricks) [inline gsa/src/web/pages/tlscertificates/component.js]: Does not make much sense to use the action creator function here to load the tls cert. It even would crash if a second request is already running.

File: gsa/src/web/pages/tlscertificates/component.js [hunk 4]
```diff
@@ -55,36 +54,38 @@ class TlsCertificateComponent extends React.Component {
...
-      username,
-    });
-
-    onDownloaded({
-      filename,
-      mimetype: 'application/x-x509-ca-cert',
-      data: create_pem_certificate(certificate),
+    const {detailsExportFileName, gmp, username, onDownloaded} = this.props;
+
+    gmp.tlscertificate.get({id: cert.id}).then(response => {
```
Comment (bjoernricks) [inline gsa/src/web/pages/tlscertificates/component.js]:
```suggestion
    return gmp.tlscertificate.get({id: cert.id}).then(response => {
```

File: gsa/src/web/pages/tlscertificates/component.js [hunk 5]
```diff
@@ -55,36 +54,38 @@ class TlsCertificateComponent extends React.Component {
...
+        extension: 'pem', // this gets overwritten to .cer in chrome
+        fileNameFormat: detailsExportFileName,
+        id,
+        modificationTime,
+        resourceName: name,
+        resourceType: entityType,
+        username,
+      });
+
+      onDownloaded({
```
Comment (bjoernricks) [inline gsa/src/web/pages/tlscertificates/component.js]:
```suggestion
      return onDownloaded({
```

File: gsa/src/web/pages/tlscertificates/component.js (hunk 4 above)
Comment (swaterkamp) [reply gsa/src/web/pages/tlscertificates/component.js]: Indeed, I forgot about that!

**Pull request merged**